cmake/modules/kconfig.cmake

```diff
@@ -113,7 +113,14 @@ foreach(module_name ${ZEPHYR_MODULE_NAMES})
     list(APPEND
          ZEPHYR_KCONFIG_MODULES_DIR
          "ZEPHYR_${MODULE_NAME_UPPER}_KCONFIG=${ZEPHYR_${MODULE_NAME_UPPER}_KCONFIG}"
-  )
+    )
+  endif()
+
+  if(KCONFIG_NAMESPACE STREQUAL "SB_CONFIG" AND SYSBUILD_${MODULE_NAME_UPPER}_KCONFIG)
+    list(APPEND
+         ZEPHYR_KCONFIG_MODULES_DIR
+         "SYSBUILD_${MODULE_NAME_UPPER}_KCONFIG=${SYSBUILD_${MODULE_NAME_UPPER}_KCONFIG}"
+    )
```

Comment:
can you place this in https://github.com/zephyrproject-rtos/zephyr/blob/main/share/sysbuild/cmake/modules/sysbuild_kconfig.cmake as we try to keep this file free of sysbuild related knowledge and instead have sysbuild knowledge in the `sysbuild_kconfig.cmake` file.